source/common/router/route_config_update_receiver_impl.cc

```diff
@@ -63,7 +63,10 @@ bool RouteConfigUpdateReceiverImpl::onRdsUpdate(const Protobuf::Message& rc,
   for (const auto& vhost : new_route_config->virtual_hosts()) {
     rds_virtual_hosts.emplace(vhost.name(), vhost);
```

Comment:
let's avoid this copy by moving hosts out of new_route_config, the rebuild is just simply append existing vhds_virtual_hosts to existing rds_v-hosts, which could be also be done 0-copy through move.

Reply:
I don't think to mess with the VHDS handling here. Both new_route_config and rds_virtual_hosts are used to rebuild vhosts in onVhdsUpdate. new_route_config is also cached for other potential uses some place else.

Reply:
We'll look into avoiding vhost copy in a follow up.